File: test-suite-python/src/test/python/micronaut/security/docs/jwtencryption/RSAOAEPEncryptionConfiguration.py
# tag::clazz[]
from typing import Annotated

from com.nimbusds.jose import EncryptionMethod, JWEAlgorithm
from jakarta.inject import Named, Singleton
from java.security.interfaces import RSAPrivateKey, RSAPublicKey
from micronaut.context.annotation import Requires, Value
from micronaut.security.token.jwt.encryption.rsa import RSAEncryptionConfiguration

from .KeyPairProvider import KeyPairProvider
# end::clazz[]


@Requires(property="spec.name", value="RSAOAEPEncryptionTest")
# tag::clazz[]

@Named("generator")  # <1>
@Singleton
class RSAOAEPEncryptionConfiguration(RSAEncryptionConfiguration):

    def __init__(self, pemPath: Annotated[str, Value("${pem.path}")]):
        self.rsaPrivateKey = None
        self.rsaPublicKey = None
        self.jweAlgorithm = JWEAlgorithm.RSA_OAEP_256
        self.encryptionMethod = EncryptionMethod.A128GCM
        keyPair = KeyPairProvider.key_pair(pemPath)
        if keyPair.isPresent():
            self.rsaPublicKey = keyPair.get().getPublic()
            self.rsaPrivateKey = keyPair.get().getPrivate()

    def getPublicKey(self) -> RSAPublicKey:
        return self.rsaPublicKey

    def getPrivateKey(self) -> RSAPrivateKey:
        return self.rsaPrivateKey

    def getJweAlgorithm(self) -> JWEAlgorithm:
        return self.jweAlgorithm

    def getEncryptionMethod(self) -> EncryptionMethod:
        return self.encryptionMethod
# end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/security/docs/jwtencryption/RSAOAEPEncryptionConfiguration.kt
package io.micronaut.security.docs.jwtencryption

//tag::clazz[]
import com.nimbusds.jose.EncryptionMethod
import com.nimbusds.jose.JWEAlgorithm
import io.micronaut.context.annotation.Requires
import io.micronaut.context.annotation.Value
import io.micronaut.security.token.jwt.encryption.rsa.RSAEncryptionConfiguration
import jakarta.inject.Named
import jakarta.inject.Singleton
import java.security.interfaces.RSAPrivateKey
import java.security.interfaces.RSAPublicKey
//end::clazz[]

@Requires(property = "spec.name", value = "RSAOAEPEncryptionTest")
//tag::clazz[]

@Named("generator") // <1>
@Singleton
class RSAOAEPEncryptionConfiguration(@Value("\${pem.path}") pemPath: String) : RSAEncryptionConfiguration {

    private var rsaPrivateKey: RSAPrivateKey? = null
    private var rsaPublicKey: RSAPublicKey? = null
    private val jweAlgorithm = JWEAlgorithm.RSA_OAEP_256
    private val encryptionMethod = EncryptionMethod.A128GCM

    init {
        val keyPair = KeyPairProvider.keyPair(pemPath)
        if (keyPair.isPresent) {
            rsaPublicKey = keyPair.get().public as RSAPublicKey
            rsaPrivateKey = keyPair.get().private as RSAPrivateKey
        }
    }

    override fun getPublicKey(): RSAPublicKey? = rsaPublicKey

    override fun getPrivateKey(): RSAPrivateKey? = rsaPrivateKey

    override fun getJweAlgorithm(): JWEAlgorithm = jweAlgorithm

    override fun getEncryptionMethod(): EncryptionMethod = encryptionMethod
}
//end::clazz[]
